events/janus_gelfevh.c

```diff
@@ -0,0 +1,617 @@
+		if(!item || !item->value || !janus_is_true(item->value)) {
+			JANUS_LOG(LOG_WARN, "GELF event handler disabled (Janus API)\n");
+			goto done;
+		}
+		item = janus_config_get(config, config_general, janus_config_type_item, "backend");
+		if(!item || !item->value) {
+			JANUS_LOG(LOG_WARN, "Missing or invalid backend\n");
+			goto done;
+		}
+		backend = g_strdup(item->value);
+		item = janus_config_get(config, config_general, janus_config_type_item, "port");
+		if(!item || !item->value) {
+			JANUS_LOG(LOG_WARN, "Missing or invalid port\n");
+			goto done;
+		}
+		port = g_strdup(item->value);
+		item = janus_config_get(config, config_general, janus_config_type_item, "protocol");
+		if (item && item->value) {
+			if (strcasecmp(item->value, "udp") == 0){
+				janus_gelfevh_socket_type = 2;
+			}
+		}
+		item = janus_config_get(config, config_general, janus_config_type_item, "max_message_len");
+		if (item && item->value) {
+			int mml = atoi(item->value);
+			max_gelf_msg_len = mml;
+		}
+		/* Which events should we subscribe to? */
+		item = janus_config_get(config, config_general, janus_config_type_item, "events");
+		if(item && item->value)
+			janus_events_edit_events_mask(item->value, &janus_gelfevh.events_mask);
+		/* Compact, so no spaces between separators */
+		json_format = JSON_COMPACT | JSON_PRESERVE_ORDER;
+
+		/* Check if we need any compression */
+		item = janus_config_get(config, config_general, janus_config_type_item, "compress");
+		if(item && item->value && janus_is_true(item->value)) {
+			compress = TRUE;
+			item = janus_config_get(config, config_general, janus_config_type_item, "compression");
+			if(item && item->value) {
+				int c = atoi(item->value);
+				if(c < 0 || c > 9) {
+					JANUS_LOG(LOG_WARN, "Invalid compression factor '%d', falling back to '%d'...\n", c, compression);
+				} else {
+					compression = c;
+				}
+			}
+		}
+		/* Done */
+		enabled = TRUE;
+	}
+done:
+	janus_config_destroy(config);
+	config = NULL;
+	if(!enabled) {
+		JANUS_LOG(LOG_FATAL, "Gelf event handler not enabled/needed, giving up...\n");
+		return -1;	/* No point in keeping the plugin loaded */
+	}
+	JANUS_LOG(LOG_VERB, "Gelf event handler configured: %s:%s\n", backend, port);
+
+	/* Initialize the events queue */
+	events = g_async_queue_new_full((GDestroyNotify) janus_gelfevh_event_free);
+	janus_mutex_init(&evh_mutex);
+
+	g_atomic_int_set(&initialized, 1);
+
+	/* Launch the thread that will handle incoming events */
+	GError *error = NULL;
+	handler_thread = g_thread_try_new("janus gelfevh handler", janus_gelfevh_handler, NULL, &error);
+	if(error != NULL) {
+		g_atomic_int_set(&initialized, 0);
+		JANUS_LOG(LOG_ERR, "Got error %d (%s) trying to launch the GelfEventHandler handler thread...\n", error->code, error->message ? error->message : "??");
+		return -1;
+	}
+	JANUS_LOG(LOG_INFO, "%s initialized!\n", JANUS_GELFEVH_NAME);
+	janus_gelfevh_connect();
```

Comment:
I think the connect should happen before spawning the thread, and have a return value, so that if connecting fails you can return a failure from the init too. Unless you plan to reconnect on a regular basis when an initial connection fails (e.g., Janus started before Graylog): "regular basis" here may also be "if we're not connected when trying to send a message, try connecting first"

Reply:
I did push this change, but I noticed that it will only work if tcp is chosen, UDP will be considered open :D